Makefile

```diff
@@ -27,8 +42,22 @@ darwin_intel:
 darwin_m1:
 	go run make.go -v DarwinM1
 
-linux:
-	go run make.go -v linux
+linux: tcpsnoop.bpf.o
+	$(GOFLAGS) go run make.go -v linux
+
+$(OUTPUT) $(OUTPUT)/libbpf:
+	@mkdir -p $@
+
+%.bpf.o: $(mkfile_dir)vql/linux/tcpsnoop/%.bpf.c $(LIBBPF_OBJ) | $(OUTPUT)
+	@$(CLANG) $(CFLAGS) -target bpf -D__TARGET_ARCH_$(ARCH)	      \
+		     -I./ $(INCLUDES) -c $(filter %.c,$^) -o $(mkfile_dir)vql/linux/tcpsnoop/$@ && \
+	$(LLVM_STRIP) -g $(mkfile_dir)vql/linux/tcpsnoop/$@
+
+$(LIBBPF_OBJ): $(wildcard $(LIBBPF_SRC)/src/*.[ch]) | $(OUTPUT)/libbpf
```

Comment:
I'm fine distributing libbpf.a itself, however if I were some external user having to download some binary package would make me slightly uneasy. But I guess it's feasible...

Reply:
Another thing regarding the bpf object could be to utilize go generate and also ship the resulting .o file. That way we won't impose the burden of people just wanting to use velociraptor to have to install clang et al but could simply use the compiled-object. And for the more tech-savvy we'd also be shipping the .bpf.c file in case they want to make modifications and ensure what code they are loading into their kernel. For more info on go generate: https://go.dev/blog/generate